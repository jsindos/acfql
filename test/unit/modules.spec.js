/* global expect, describe, it */

const ACFStore = require('../../generate/acfstore')
const { generateTemplate: generateIndexTemplate } = require('../../src/modules/CustomFields/connectors/index.template.js')
const { generateTemplate: generateGetFieldTemplate } = require('../../src/modules/CustomFields/connectors/getField.template.js')

describe('generate module files', () => {
  it('generateIndexTemplate', () => {
    const a = new ACFStore()
    a.addFieldGroup('Apple Information')
    a.addField('cultivar', 'Cultivar', 'Apple Information')
    a.addField('price', 'Price', 'Apple Information')
    a.addFieldGroup('Other Information')
    a.addField('other', 'Other', 'Other Information')
    expect(generateIndexTemplate(a.customPostTypes, a.fieldGroups)).toMatchSnapshot()
  })

  it('generateFieldTemplate', () => {
    const a = new ACFStore()
    a.addFieldGroup('Apple Information')
    a.addField('cultivar', 'Cultivar', 'Apple Information')
    a.addField('price', 'Price', 'Apple Information')
    a.addFieldGroup('Other Information')
    a.addField('other', 'Other', 'Other Information')
    const templates = generateGetFieldTemplate(a.customPostTypes, a.fieldGroups)
    expect(templates.find(t => t.fileName === 'getAppleInformationCultivar').template).toMatchSnapshot()
  })
})
