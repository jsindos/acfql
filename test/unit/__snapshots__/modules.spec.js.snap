// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate module files generateFieldTemplate 1`] = `
"const getField = require('./utility/getField')

module.exports = (Post, Postmeta) => {
  return function ({ postId }) {
    return Postmeta.findOne({
      where: {
        meta_key: 'cultivar',
        post_id: postId
      }
    }).then(postMeta => postMeta && postMeta.meta_value)
  }
}
"
`;

exports[`generate module files generateIndexTemplate 1`] = `
"// AppleInformation fields
const getAppleInformationCultivar = require('./getAppleInformationCultivar')
const getAppleInformationPrice = require('./getAppleInformationPrice')

// OtherInformation fields
const getOtherInformationOther = require('./getOtherInformationOther')


function Connectors ({ Post, Postmeta }) {
  return {
    
    getAppleInformationCultivar: getAppleInformationCultivar(Post, Postmeta),
    getAppleInformationPrice: getAppleInformationPrice(Post, Postmeta),
    getOtherInformationOther: getOtherInformationOther(Post, Postmeta)
  }
}

module.exports = Connectors
"
`;
